V13: Webhook all the things

warrenbuckley opened the pull request:
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes <!-- link to the issue here! -->

### Description
This adds in more WebHooks into the CMS as the are plenty of Notifications/Events raised by Umbraco that people may want to be notified by as a webhook posted externally to a third party system.

This PR adds in a lot of WebHook Event Types but they are not automatically added by default and would require you to update the Webhook Collection in a composer like so, where an implementor can pick and choose what collection of WebHooks they want to add.

```csharp
// Pick and choose from the collection of Webhooks
builder.WebhookEvents()
            .AddContentWebhooks()
            .AddDataTypeWebhooks()
            .AddDictionaryWebhooks()
            .AddDomainWebhooks()
            .AddLanguageWebhooks()
            .AddMediaWebhooks()
            .AddMemberWebhooks()
            .AddMemberTypeWebhooks()
            .AddPackageWebhooks()
            .AddPublicAccessWebhooks()
            .AddRelationWebhooks()
            .AddScriptWebhooks()
            .AddStylesheetWebhooks()
            .AddTemplateWebhooks()
            .AddUserWebhooks();
            
// Or go nuts and add them all
builder.WebhookEvents()
            .AddAllAvailableWebhooks();
```


## Test Notes
* Enable a webhook from the long list and ensure the correct information is sent
* Use a site such as https://webhook.site to test the payloads

## Updated
### Content
- [x] ContentCopied 
- [x] ContentDeletedBlueprint
- [x] ContentDeletedVerions
- [x] ContentDeleted
- [x] ContentEmptiedRecycleBin
- [x] ContentMovedToRecycleBin
- [x] ContentMoved
- [x] ContentPublished
- [x] ContentRolledBack
- [x] ContentSavedBluePrint
- [x] ContentSaved
- [x] ContentSorted
- [x] ContentUnpublished

### DataType
- [x] DataTypeDeleted
- [x] DataTypeMoved
- [x] DataTypeSaved

### Dictionary
- [x] DictionaryItemDeleted
- [x] DictionaryItemSaved

### Domain
- [x] DomainDeleted
- [x] DomainSaved

### Language
- [x] LanguageDeleted
- [x] LanguageSaved

### Media
- [x] MediaDeleted
- [x] MediaEmptiedRecycleBin
- [x] MediaMovedToRecycleBin
- [x] MediaMoved
- [x] MediaSaved

### MediaType
- [x] MediaTypeChanged
- [x] MediaTypeDeleted
- [x] MediaTypeMoved
- [x] MediaTypeSaved

### Member
- [x] MemberDeleted
- [x] MemberSaved
- [x] MemberGroupDeleted
- [x] MemberGroupSaved
- [x] AssignedMemberRoles
- [x] RemovedMemberRoles
- [x] ExportedMember

### Member Type
- [x] MemberTypeSaved
- [x] MemberTypeDeleted
- [x] MemberTypeChanged
- [x] MemberTypeMoved

### Package
- [x] ImportedPackage

### Public Access
- [x] PublicAccessEntryDeleted
- [x] PublicAccessEntrySaved

### Relation
- [x] RelationDeleted
- [x] RelationSaved
- [x] RelationTypeDeleted
- [x] RelationTypeSaved

### Script
- [x] ScriptDeleted
- [x] ScriptSaved

### Stylesheet
- [x] StylesheetDeleted
- [x] StylesheetSaved

### Template
- [x] PartialViewDeleted
- [x] PartialViewSaved
- [x] TemplateDeleted
- [x] TemplateSaved

### User
- [x] UserDeleted
- [x] UserSaved
- [x] UserLocked
- [x] UserUnlocked
- [x] UserLoginFailed
- [x] UserLoginSuccess
- [x] UserLogoutSuccess
- [x] UserGroupDeleted
- [x] UserGroupSaved
- [x] UserPasswordChanged
- [x] UserPasswordReset
- [x] UseForgottonPasswordRequested
- [x] UserForgottonPasswordChanged
- [x] UserLoginRequiresVerification
- [x] UserTwoFactorRequested
- [x] AssignedUserGroupPermissions



<!-- 
    A description of the changes proposed in the pull-request and how to test these changes.

    The most successful pull requests usually look a like this:

    * Fill in this template with details: what did you do, why did you do it, how can we test the changes?
    * Include screenshots and animated GIFs in your pull request whenever possible.
    * Unit tests, while optional are awesome, thank you!

    While these are guidelines and not strict requirements, they really help us evaluate your PR quicker.
-->


<!-- Thanks for contributing to Umbraco CMS! -->

mikecp commented on src/Umbraco.Core/Webhooks/WebhookEventCollectionBuilder.cs:
```diff
@@ -78,4 +71,28 @@ private void RegisterTypes(IServiceCollection services)
+        Append<LanguageDeletedWebhookEvent>();
+        Append<LanguageSavedWebhookEvent>();
+        return this;
+    }
```
And then a method `WebhookAllTheThings `that calls all 3 at once ?? 😁😇

warrenbuckley replied:
haha will gladly add a `WebHookAllTheThings` method once I'm done 😝

warrenbuckley commented:
@mikecp OK gonna pause & catch up on that thread over there...

warrenbuckley commented:
## My own notes


Member
Member Type


User


Content


ContentCopied
ContentMoved
ContentDeleted
Published
Saved
Sorted
Unpublished
ContentDeletedBlueprint
ContentRolledBack
ContentSavedBluePrint
ContentEmptiedRecycleBin
ContentMovedToRecycleBin

warrenbuckley commented:
@Zeegaan before I wrap this up and do anymore, is this likely this PR will be accepted?
I know I should have checked first 🙈

Zeegaan commented:
Depends on how well you can argue your use case to @bergmania I think 🙈

warrenbuckley commented:
@bergmania let me know what you think matey 😄

bergmania commented:
@warrenbuckley, I like the idea, just go for it 💪 🏃

Zeegaan commented:
That looks very unintentional, we should just be using the normal JsonSerializer, but I guess something wrong there.
Maybe because we're doing string content ? 🤔 
We're doing the serialization here: https://github.com/umbraco/Umbraco-CMS/blob/v13/dev/src/Umbraco.Infrastructure/BackgroundJobs/Jobs/WebhookFiring.cs#L94

warrenbuckley commented:
> That looks very unintentional, we should just be using the normal JsonSerializer, but I guess something wrong there. Maybe because we're doing string content ? 🤔 We're doing the serialization here: https://github.com/umbraco/Umbraco-CMS/blob/v13/dev/src/Umbraco.Infrastructure/BackgroundJobs/Jobs/WebhookFiring.cs#L94

@Zeegaan looks like the Payload parameter in the method of `SendRequestAsync` is already a JSON string and basically gets escaped, hence the weird formatting.

Zeegaan commented:
We should fix that then 🙈

warrenbuckley commented:
@Zeegaan I haven't forgotten about this PR, just been mega busy _(not much lunchtime hacking or evening hacks this week)_

warrenbuckley commented:
@Zeegaan or @bergmania any smart ideas on how I could ignore certain properties for JSON serialistion for some of these Webhook for Notifications.

As I doubt people will be too happy if password submitted as a webhook payload for members and users.

My initial train of thought was perhaps to use `[JsonIgnore]` attribute, but my worry is that if you are using these notifications and serializing them and using them anywhere else, it could potentially break other things.

So my next train of thought was to perhaps use a new attribute `[WebhookIgnore]` that you would put on a property that would work in a similar way, but use JSON serialization settings or something to ensure only these ones are excluded when we serialise for the webhook payload.

Any smart ideas or other thoughts ?

Zeegaan commented:
It can be achieved  if they were to create their own webhooks, and then overriding the `ConvertEntityToRequestPayload` method. 🤷

I don't  see a smarter way to do it from our point of view at the momemt 🙈

warrenbuckley commented:
@Zeegaan Would it make sense to have a similar override of `ConvertEntityToRequestPayload` on the simpler `WebhookEventBase` that takes in just a Notification so then could create an anonymous object which does not include the password stuff for Users and Members

Zeegaan commented:
Not sure why that would be needed for the one with just Notification, the point of that one is that we do not need to send a request object theoretically 🙈 
The other ones could "just" use the other one.

Maybe I'm not seeing the use case clearly in my head, so if you show me an example of how you would use it 🤔

warrenbuckley commented:
```csharp
using Microsoft.Extensions.Options;
using Umbraco.Cms.Core.Configuration.Models;
using Umbraco.Cms.Core.Notifications;
using Umbraco.Cms.Core.Services;
using Umbraco.Cms.Core.Sync;

namespace Umbraco.Cms.Core.Webhooks.Events.Member;

[WebhookEvent("Member Saved")]
public class MemberSavedWebhookEvent : WebhookEventBase<MemberSavedNotification>
{
    public MemberSavedWebhookEvent(
        IWebhookFiringService webhookFiringService,
        IWebhookService webHookService,
        IOptionsMonitor<WebhookSettings> webhookSettings,
        IServerRoleAccessor serverRoleAccessor)
        : base(webhookFiringService, webHookService, webhookSettings, serverRoleAccessor)
    {
    }

    public override string Alias => "memberSaved";
}
```

## Problem
The other base class depends that the TEntity inherits from `IContentBase` so you can't shape the JSON payload for a notification, which you may want to exclude certain properties as some of the data can be not that useful to receive in a webhook.

warrenbuckley commented:
Right think I have got them all, well most of the notifications.
I need to take another pass over them to make sure the data sent is useful before handing over to the PR or HQ Core team

Zeegaan commented:
Ah I see the idea, yes I'm not opposed to being able to overwrite the output (I'd actually kinda expect to be able to do that 🙈)

warrenbuckley commented:
@Zeegaan have you seen this error at all, when working with WebHooks ?

```
System.AggregateException: One or more errors occurred. (One or more errors occurred. (String or binary data would be truncated in table 'D:\CODE\PERSONAL\UMBRACO-CMS\SRC\UMBRACO.WEB.UI\UMBRACO\DATA\UMBRACO.MDF.dbo.umbracoWebhookRequest', column 'requestObject'. Truncated value: '{"CreateTemplateForContentType":true,"ContentTypeAlias":"home","SavedEntities":[{"$id":"1","MasterTe'.
The statement has been terminated.))
 ---> System.AggregateException: One or more errors occurred. (String or binary data would be truncated in table 'D:\CODE\PERSONAL\UMBRACO-CMS\SRC\UMBRACO.WEB.UI\UMBRACO\DATA\UMBRACO.MDF.dbo.umbracoWebhookRequest', column 'requestObject'. Truncated value: '{"CreateTemplateForContentType":true,"ContentTypeAlias":"home","SavedEntities":[{"$id":"1","MasterTe'.
The statement has been terminated.)
 ---> Microsoft.Data.SqlClient.SqlException (0x80131904): String or binary data would be truncated in table 'D:\CODE\PERSONAL\UMBRACO-CMS\SRC\UMBRACO.WEB.UI\UMBRACO\DATA\UMBRACO.MDF.dbo.umbracoWebhookRequest', column 'requestObject'. Truncated value: '{"CreateTemplateForContentType":true,"ContentTypeAlias":"home","SavedEntities":[{"$id":"1","MasterTe'.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at Microsoft.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at Microsoft.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at Microsoft.Data.SqlClient.SqlDataReader.get_MetaData()
   at Microsoft.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean isAsync, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry, String method)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method)
   at Microsoft.Data.SqlClient.SqlCommand.ExecuteScalar()
   at StackExchange.Profiling.Data.ProfiledDbCommand.ExecuteScalar() in C:\projects\dotnet\src\MiniProfiler.Shared\Data\ProfiledDbCommand.cs:line 315
   at Umbraco.Cms.Infrastructure.Persistence.FaultHandling.FaultHandlingDbCommand.<ExecuteScalar>b__36_0() in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Infrastructure\Persistence\FaultHandling\RetryDbConnection.cs:line 185
   at Umbraco.Cms.Infrastructure.Persistence.FaultHandling.FaultHandlingDbCommand.<>c__DisplayClass38_0`1.<Execute>b__0() in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Infrastructure\Persistence\FaultHandling\RetryDbConnection.cs:line 193
   at Umbraco.Cms.Infrastructure.Persistence.FaultHandling.RetryPolicy.ExecuteAction[TResult](Func`1 func) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Infrastructure\Persistence\FaultHandling\RetryPolicy.cs:line 179
   at Umbraco.Cms.Infrastructure.Persistence.FaultHandling.FaultHandlingDbCommand.Execute[T](Func`1 f) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Infrastructure\Persistence\FaultHandling\RetryDbConnection.cs:line 190
   at Umbraco.Cms.Infrastructure.Persistence.FaultHandling.FaultHandlingDbCommand.ExecuteScalar() in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Infrastructure\Persistence\FaultHandling\RetryDbConnection.cs:line 185
   at System.Data.Common.DbCommand.ExecuteScalarAsync(CancellationToken cancellationToken)
   at Umbraco.Cms.Infrastructure.Persistence.FaultHandling.FaultHandlingDbCommand.Execute[T](Func`1 f) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Infrastructure\Persistence\FaultHandling\RetryDbConnection.cs:line 190
   at Umbraco.Cms.Infrastructure.Persistence.FaultHandling.FaultHandlingDbCommand.ExecuteScalar() in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Infrastructure\Persistence\FaultHandling\RetryDbConnection.cs:line 185
   at System.Data.Common.DbCommand.ExecuteScalarAsync(CancellationToken cancellationToken)
--- End of stack trace from previous location ---
   at NPoco.Database.ExecutionHookAsync[T](Func`1 action)
   at NPoco.Database.ExecuteScalarHelperAsync(DbCommand cmd)
   at NPoco.Database.InsertAsyncImp[T](PocoData pocoData, String tableName, String primaryKeyName, Boolean autoIncrement, T poco, Boolean sync)
   at Umbraco.Cms.Infrastructure.Persistence.Repositories.Implement.WebhookRequestRepository.CreateAsync(WebhookRequest webhookRequest) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Infrastructure\Persistence\Repositories\Implement\WebhookRequestRepository.cs:line 33
   at Umbraco.Cms.Core.Services.WebhookRequestService.CreateAsync(Guid webhookKey, String eventAlias, Object payload) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Core\Services\WebhookRequestService.cs:line 33
   at Umbraco.Cms.Infrastructure.Services.Implement.WebhookFiringService.FireAsync(Webhook webhook, String eventAlias, Object payload, CancellationToken cancellationToken) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Infrastructure\Services\Implement\WebhookFiringService.cs:line 13
   at Umbraco.Cms.Core.Webhooks.WebhookEventBase`1.ProcessWebhooks(TNotification notification, IEnumerable`1 webhooks, CancellationToken cancellationToken) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Core\Webhooks\WebhookEventBase.cs:line 65
   at Umbraco.Cms.Core.Webhooks.WebhookEventBase`1.HandleAsync(TNotification notification, CancellationToken cancellationToken) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Core\Webhooks\WebhookEventBase.cs:line 96
   at Umbraco.Cms.Core.Events.INotificationAsyncHandler`1.HandleAsync(IEnumerable`1 notifications, CancellationToken cancellationToken) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Core\Events\INotificationAsyncHandler.cs:line 37
   at Umbraco.Cms.Core.Events.EventAggregator.PublishCoreAsync[TNotification](IEnumerable`1 allHandlers, IEnumerable`1 notifications, CancellationToken cancellationToken) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Core\Events\EventAggregator.Notifications.cs:line 122
   at Umbraco.Cms.Core.Events.NotificationAsyncHandlerWrapperImpl`1.HandleAsync[TNotification,TNotificationHandler](IEnumerable`1 notifications, CancellationToken cancellationToken, ServiceFactory serviceFactory, Func`4 publish) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Core\Events\EventAggregator.Notifications.cs:line 259
   at Umbraco.Cms.Core.Events.EventAggregator.PublishNotificationsAsync[TNotification,TNotificationHandler](IEnumerable`1 notifications, CancellationToken cancellationToken) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Core\Events\EventAggregator.Notifications.cs:line 101
ClientConnectionId:f1d7d9d9-a977-4cbb-86a0-9192bca4383c
Error Number:2628,State:1,Class:16
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.WaitAllCore(Task[] tasks, Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.WaitAll(Task[] tasks)
   at Umbraco.Cms.Core.Events.EventAggregator.Publish[TNotification,TNotificationHandler](IEnumerable`1 notifications) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Core\Events\EventAggregator.cs:line 31
   at Umbraco.Cms.Core.Events.ScopedNotificationPublisher`1.PublishScopedNotifications(IList`1 notifications) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Core\Events\ScopedNotificationPublisher.cs:line 117
   at Umbraco.Cms.Core.Events.ScopedNotificationPublisher`1.ScopeExit(Boolean completed) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Core\Events\ScopedNotificationPublisher.cs:line 94
   at Umbraco.Cms.Core.Scoping.CoreScope.HandleScopedNotifications() in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Core\Scoping\CoreScope.cs:line 253
   at Umbraco.Cms.Infrastructure.Scoping.Scope.TryFinally(Action[] actions) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Infrastructure\Scoping\Scope.cs:line 581
   --- End of inner exception stack trace ---
   at Umbraco.Cms.Infrastructure.Scoping.Scope.TryFinally(Action[] actions) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Infrastructure\Scoping\Scope.cs:line 591
   at Umbraco.Cms.Infrastructure.Scoping.Scope.RobustExit(Boolean completed, Boolean onException) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Infrastructure\Scoping\Scope.cs:line 566
   at Umbraco.Cms.Infrastructure.Scoping.Scope.DisposeLastScope() in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Infrastructure\Scoping\Scope.cs:line 502
   at Umbraco.Cms.Infrastructure.Scoping.Scope.Dispose() in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Infrastructure\Scoping\Scope.cs:line 414
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentTypeController.CreateTemplateForContentType(String contentTypeAlias, String contentTypeName) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Web.BackOffice\Controllers\ContentTypeController.cs:line 515
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentTypeController.<PostSave>b__34_2(DocumentTypeSave ctSave) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Web.BackOffice\Controllers\ContentTypeController.cs:line 435
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentTypeControllerBase`1.PerformPostSave[TContentTypeDisplay,TContentTypeSave,TPropertyType](TContentTypeSave contentTypeSave, Func`2 getContentType, Action`1 saveContentType, Action`1 beforeCreateNew) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Web.BackOffice\Controllers\ContentTypeControllerBase.cs:line 374
   at Umbraco.Cms.Web.BackOffice.Controllers.ContentTypeController.PostSave(DocumentTypeSave contentTypeSave) in D:\Code\Personal\Umbraco-CMS\src\Umbraco.Web.BackOffice\Controllers\ContentTypeController.cs:line 426
   at lambda_method784(Closure, Object, Object[])
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(ActionContext actionContext, IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Logged|12_1(ControllerActionInvoker invoker)
   ```

Zeegaan commented:
Yep, the RequestObject on Webhook request was not set to NVarcharMax, this has been fixed in https://github.com/umbraco/Umbraco-CMS/pull/15252 🙈

warrenbuckley commented:
What are your thoughts @Zeegaan & @bergmania of adding all these WebHooks by default out of the box or would you prefer people to opt in?

Zeegaan commented:
Yes I think grouping and the stuff you mentioned is almost mandatory with this one, as there is suddenly a lot of events to search through to find the one you need to use 👍

warrenbuckley commented on src/Umbraco.Core/Webhooks/WebhookEventCollectionBuilder.cs:
```diff
@@ -78,4 +85,187 @@ private void RegisterTypes(IServiceCollection services)
         return null;
     }
+
+    public WebhookEventCollectionBuilder AddAllTheWebhooks() =>
```
@ronaldbarendse point me to an example and I'll happily lift & shift it over

warrenbuckley commented:
@Zeegaan surely worth me doing a separate PR for that once this is merged

warrenbuckley commented:
Is there anything I can do to help you and the Core Collabs team with the review/testing process @mikecp or are you wanting the HQ gang such as @Zeegaan to pick this up and review it.

Just checking I am not stuck in limbo land between the two groups 😄

warrenbuckley commented:
## UX improvement
Updated PR for the EventPicker Infinite Editor so that the events are ordered by the friendly `eventName` property and implemented filtering in the same way the icon picker does.

https://github.com/umbraco/Umbraco-CMS/assets/1389894/52766262-e56e-4191-8303-c45893facb48

bjarnef commented:
Maybe we could just have a dropdown to filter the category, e.g. Content, Media, (Member), DataType, etc. which filter the event by name (or prefix) and IIRC a selection will filter other event types from the list anyway.

warrenbuckley commented:
@bjarnef yes perhaps we could, but for now I followed another UX pattern that was already in the backoffice.
Hopefully this gets looked at soon and then if you wanted to you could iterate/improve the UX of this.
